vip-cache-manager/includes/class-rest-api.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Plugin's REST API
+ *
+ * @package a8c_Cache_Manager
+ */
+
+require_once( __DIR__ . '/class-singleton.php' );
+
+/**
+ * REST API class
+ */
+class REST_API extends Singleton {
+
+	/**
+	 * API SETUP
+	 */
+	const API_NAMESPACE  = 'cache-manager/v1';
+	const ENDPOINT_PURGE = 'purge'; // purge URL(s)
+	const ENDPOINT_BAN   = 'ban'; // ban URL(s)
+	const URLS_KEY       = 'urls';
+
+	/**
+	 * PLUGIN SETUP
+	 */
+
+	/**
+	 * Register hooks
+	 */
+	protected function class_init() {
+		add_action( 'rest_api_init', array( $this, 'rest_api_init' ) );
+	}
+
+	/**
+	 * PLUGIN FUNCTIONALITY
+	 */
+
+	/**
+	 * Register API routes
+	 */
+	public function rest_api_init() {
+
+		register_rest_route( self::API_NAMESPACE, '/' . self::ENDPOINT_PURGE, array(
+			'methods'             => 'POST',
+			'callback'            => array( $this, 'purge_urls' ),
+			'permission_callback' => array( $this, 'check_secret' ),
+			'show_in_index'       => false,
+		) );
+
+		register_rest_route( self::API_NAMESPACE, '/' . self::ENDPOINT_BAN, array(
+			'methods'             => 'POST',
+			'callback'            => array( $this, 'debug_request' ),
+			'permission_callback' => array( $this, 'check_secret' ),
+			'show_in_index'       => false,
+		) );
+
+	}
+
+	/**
+	 * Testing function
+	 */
+	public function debug_request( $request ) {
```

Comment:
Looks like we need to flesh out the BAN method here

Reply:
Sure, it was just a skeleton but now that we agreed on not using it - did we? -, it can be cleaned up.

Reply:
Sure, let's remove it for now.

Reply:
Done.

Reply:
* Main class does not extend Singleton anymore.
* Ban route has been removed.
* Now using wpcom_vip_go_rest_api_request_allowed in place of shared secret.